lti_consumer/plugin/views.py

```diff
@@ -82,6 +83,45 @@ def has_block_access(user, block, course_key):
     return course_access and block_access
 
 
+@require_http_methods(["GET"])
+def initiate_course_tool_launch_endpoint(request, usage_id, lti_config_id):
+    """
+    Launch an LTI tool configured platform wide with a given course_id as the learning context.
+    """
+    try:
+        lti_config = LtiConfiguration.objects.get(config_id=lti_config_id)
+
+        if lti_config.version != lti_config.LTI_1P3:
+            raise LtiError(
+                "LTI Error: LTI 1.1 configurations not supported."
+            )
+    except (LtiError, ObjectDoesNotExist):
+        raise Http404 from exc
+
+    # Tools configured with a specific placement cannot be launched course wide
+    # Mostly a hack in place of properly modeling 'platform wide' configurations
+    if lti_config.location:
+        raise PermissionDenied
+
+    location = UsageKey.from_string(usage_id)
+    course_key = location.course_key
+
+    # Problem. How can we get the user's role for the passed in course/context?
+    # We have to use a compat layer to do this which gets messy because this isn't
+    # just needed in the LMS but also exams.
+    user_role = None
```

Comment:
this is really problematic when it comes to running this library in anything other than the LMS. We need to ask the running service for user permissions.